Content.Client/Silicons/Laws/Ui/LawDisplay.xaml.cs

```diff
@@ -1,21 +1,81 @@
+using Content.Client.Chat.Managers;
+using Content.Shared.Chat;
+using Content.Shared.Radio;
 using Content.Shared.Silicons.Laws;
 using Robust.Client.AutoGenerated;
 using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
 using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
 
 namespace Content.Client.Silicons.Laws.Ui;
 
 [GenerateTypedNameReferences]
 public sealed partial class LawDisplay : Control
 {
-    public LawDisplay(SiliconLaw prototype)
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly IChatManager _chatManager = default!;
+
+    public event Action<BaseButton.ButtonEventArgs>? OnLawAnnouncementButtonPressed;
+
+    public LawDisplay(SiliconLaw law, bool canVerbalize, HashSet<string> radioChannels)
     {
         RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        var identifier = law.LawIdentifierOverride ?? $"{law.Order}";
+        string lawIdentifier = Loc.GetString("laws-ui-law-header", ("id", identifier));
+        string lawDescription = Loc.GetString(law.LawString);
+
+        LawNumberLabel.Text = lawIdentifier;
+        LawLabel.SetMessage(lawDescription);
+
+        if (!canVerbalize) // If you can't talk, you can't state your laws...
+            return;
+
+        Button localButton = new Button
+        {
+            Text = Loc.GetString("hud-chatbox-select-channel-Local"),
+            Modulate = Color.DarkGray,
+            StyleClasses = { "chatSelectorOptionButton" },
+            MinHeight = 35,
+            MinWidth = 75,
+        };
+
+        localButton.OnPressed += _ =>
+        {
+            _chatManager.SendMessage($"{lawIdentifier}: {lawDescription}", ChatSelectChannel.Local);
+        };
+
+        LawAnnouncementButtons.AddChild(localButton);
+
+        foreach (string radioChannel in radioChannels)
+        {
+            if (!_prototypeManager.TryIndex<RadioChannelPrototype>(radioChannel, out var radioChannelProto))
+                continue;
+
+            Button radioChannelButton = new Button
+            {
```

Comment:
```suggestion
            var radioChannelButton = new Button
            {
```

Reply:
There's also a few other C# style guide corrections needing elsewhere.

Reply:
Are these issues related to what's covered in https://docs.spacestation14.io/en/getting-started/conventions?

If not, could you please link me to the appropriate style guide? Cheers